### PR TITLE
[Docathon][Remove File No.47-55]清理未暴露在 sphinx toctree 下的文档

### PR DESCRIPTION
**Pull request by hu-qi**
| 序号    | 文件或目录 |是否保留 |原因|链接|
| -------- | ------- |------- |------- | ------- | 
| 47  | docs/dev_guides/op_optimization/kernel_primitive_api/add_example_en.md  |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/add_example_en.html |
| 48  | docs/dev_guides/op_optimization/kernel_primitive_api/api_description_en.rst   |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/api_description_en.html |
| 49  | docs/dev_guides/op_optimization/kernel_primitive_api/compute_api_en.md   |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/compute_api_en.html|
| 50  | docs/dev_guides/op_optimization/kernel_primitive_api/example_en.rst  |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/example_en.html|
| 51  | docs/dev_guides/op_optimization/kernel_primitive_api/functor_api_en.md    |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/functor_api_en.html |
| 52  | docs/dev_guides/op_optimization/kernel_primitive_api/index_en.rst|否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/index_en.html |
| 53  | docs/dev_guides/op_optimization/kernel_primitive_api/io_api_en.md    |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/io_api_en.html|
| 54  | docs/dev_guides/op_optimization/kernel_primitive_api/model_example_en.md  |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/model_example_en.html |
| 55  | docs/dev_guides/op_optimization/kernel_primitive_api/reduce_example_en.md    |否    |未被引用   |https://www.paddlepaddle.org.cn/documentation/docs/en/develop/dev_guides/op_optimization/kernel_primitive_api/reduce_example_en.html |

### CODE REVIEW

**Comment by sunzhongkai588** [comment]
@hu-qi  这部分应该是得暴露在官网的，应该是这里的引用链接没写对，导致没有暴露
https://github.com/PaddlePaddle/docs/blob/446653d0927d18510695c74e4f7b9b766d171e72/docs/dev_guides/index_en.rst#L9

正确的链接应该是` <./op_optimization/kernel_primitive_api/index_en.html>`

**Comment by hu-qi** [comment]
> @hu-qi 这部分应该是得暴露在官网的，应该是这里的引用链接没写对，导致没有暴露
> 
> https://github.com/PaddlePaddle/docs/blob/446653d0927d18510695c74e4f7b9b766d171e72/docs/dev_guides/index_en.rst#L9
> 
> 正确的链接应该是` <./op_optimization/kernel_primitive_api/index_en.html>`

不知道这么修改对不对，您看看。

File: docs/dev_guides/index_en.rst [hunk 1]
```diff
@@ -6,13 +6,13 @@ We very much welcome you to participate in the construction of the paddle. The f
 
 Similarly, if you feel that this document is missing, or that the description is unclear, we also welcome you to contribute to this series of documents.
 
-- `Kernel Primitives API <./kernel_primitive_api/index_cn.html>`_ : Introduce the block-level CUDA functions provided by PaddlePaddle to speed up operator development.
+- `Kernel Primitives API <./op_optimization/kernel_primitive_api/index_cn.html>`_ : Introduce the block-level CUDA functions provided by PaddlePaddle to speed up operator development.
```
Comment (sunzhongkai588) [inline docs/dev_guides/index_en.rst]:
```suggestion
- `Kernel Primitives API <./op_optimization/kernel_primitive_api/index_en.html>`_ : Introduce the block-level CUDA functions provided by PaddlePaddle to speed up operator development.
```

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Pull request merged**